lib/features/label-editing/cmd/UpdateLabelHandler.js

```diff
@@ -23,7 +24,30 @@ var NULL_DIMENSIONS = {
 /**
  * A handler that updates the text of a BPMN element.
  */
-export default function UpdateLabelHandler(modeling, textRenderer) {
+export default function UpdateLabelHandler(modeling, textRenderer, bpmnFactory) {
+
+  /**
+   * Creates an empty `diLabel` attribute for embedded labels.
+   *
+   * @param {djs.model.Base} element
+   * @param {string} text
+   */
+  function ensureInternalLabelDi(element, text) {
+    if (isLabelExternal(element)) {
+      return;
+    }
+
+    var di = getDi(element);
+
+    if (text && !di.label) {
+      di.label = bpmnFactory.create('bpmndi:BPMNLabel');
```

Comment:
What I'm wondering here: How much more valid is a label without bounds? Did we really advance our BPMN support here?

Proposal: Set the label width/height/x/y to the element bounds (or position for participant / sub-process).

Reply:
My proposal obviously depends on what the BPMN spec says. How do other vendors handle this, what do they expect in terms of embedded DI labels?

Reply:
This is actually defined in the spec:

> The depiction resolution tables provided below exemplify default label positions for
BPMNShape kinds (Section 12.3.2) and BPMNEdge kinds (Section 12.3.3) if no BPMNLabel bounds are provided.

The default position for Labels without bounds is the middle of the shape

Reply:
cf. https://www.omg.org/spec/BPMN/2.0/PDF#page=411

Reply:
@nikku Check out the issue: https://github.com/bpmn-io/bpmn-js/issues/1540

Reply:
Thanks for clarifying. You're awesome folks :tada:!